Scripts/softclip_graph.py

```diff
@@ -0,0 +1,224 @@
+from datetime import datetime, date
+import pysam
+import argparse
+import os
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--bam', '-b', required=True, type=str, help='Path to bam file.')
+parser.add_argument('--output', '-o', required=False, type=str, help='Path to output folder.')
+parser.add_argument('--log', '-l', required=False, default=False, type=bool,
+                    help='Bool specifying if logfile should be made.')
+parser.add_argument('--region', '-r', required=False, default='', type=str, help='String specifying the region in '
+                                                                                 'format: "chr#:start-stop". use '
+                                                                                 'chr#:0-0 for whole chromosome.')
+parser.add_argument('--name', '-n', required=False, default='output', type=str, help='Name for the project. This is'
+                                                                                     'the name of the output file.')
+args = parser.parse_args()
+
+
+def fetch_reads():
+    """ The fetch_reads function fetches the reads from the bam file
+
+    :return reads: Pysam object containing read information
+    """
+    bamfile = pysam.AlignmentFile(args.bam, 'rb')
+
+    if args.region == 'all':
+        reads = bamfile.fetch()
+
+    else:
+        chromosome = args.region.split(':')[0].replace('chr', '')
+        start = args.region.split(':')[1].split('-')[0]
+        end = args.region.split(':')[1].split('-')[1]
+
+        if start == '0' and end == '0':
+            reads = bamfile.fetch(chromosome)
+
+        else:
+            reads = bamfile.fetch(chromosome, int(start), int(end))
+
+    return reads
+
+
+def get_softclipdata(reads):
+    """ The get_softclipdata function is a function that retrieves the position of all softclip bases. It also keeps
+    track of the number of the total number of reads and the number of reads it could not interpret.
+
+    :param reads: A pysam object containing all the reads with their information.
+    :return softclipdata: A dictionary with positions and the number of "normal" bases and softclip bases.
+    :read_data: a list containing the total number of reads and the reads it could not interpret.
+    """
+    softclipdata = {}
+    read_data = [0, 0, 0]
+
+    for read in reads:
+        if not read.is_unmapped and len(read.positions) != 0:
+            read_start = true_start(read.cigar, read.positions[0])
+            regions = softclip_regions(read_start, read.cigar)
+
+            if len(regions) != 0:
+                for region in regions:
+                    softclipdata = update_softclipdata(region, softclipdata)
```

Comment:
Een for loop over een lege lijst doet niets, dus je hoeft dat niet extra te checken.